### PR TITLE
auto-detect missing language server and prompt to install 

### PR DESCRIPTION
**Pull request by amyXia1994**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [X] Y 

partly solved #22 
re #18 

#### 2. What is the scope of this PR (e.g. component or file name):
src/extension.ts
src/install.ts
src/kclStatus.ts

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

auto-detect missing language server and prompt to install. The latest binary will be installed to ~/.kcl/kpm/bin/kcl-language-server

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [X] Y 

The old python version language-server will be deprecated. If no rust-ver language-server found, an error message will prompt to ask for installation.

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: src/install.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	switch(selected) {
+		case 'Install':
+			return installLanguageServer();
+		default:
+			return;
+	}
+}
+
+
+export async function installLanguageServer(): Promise<string | undefined> {
```
Comment (Peefy) [inline src/install.ts:32]: It is best to add CI unit tests for different platforms here to ensure successful download.

Comment (amyXia1994) [reply src/install.ts:39]: added unit test and e2e tests.

File: src/install.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+import * as os from 'os';
+import * as path from 'path';
+import axios from 'axios';
+import * as fs from 'fs';
+import { Readable } from 'node:stream';
+import * as shelljs from 'shelljs';
+
+const KCL_PATH = path.join(os.homedir(), '.kcl');
+const KPM_PATH = path.join(KCL_PATH, 'kpm');
+const KPM_BIN_PARTH = path.join(KPM_PATH, 'bin');
```
Comment (Peefy) [inline src/install.ts:12]: s/KPM_BIN_PARTH/KPM_BIN_PATH

Comment (amyXia1994) [reply src/install.ts]: fixed

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**